projects/admin/src/app/service/budget-total.service.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * RERO ILS UI
+ * Copyright (C) 2020 RERO
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, version 3 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { Injectable } from '@angular/core';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class BudgetTotalService {
+
+  /**
+   * Budget Total
+   */
+  private budgetTotal = 0;
+
+  /**
+   * Total
+   * @return number
+   */
+  getTotal() {
+    return this.budgetTotal;
+  }
+
+  /**
+   * Add amount
+   * @param amount - number
+   * @return this
+   */
+  addAmount(amount: number) {
+    this.budgetTotal = this.budgetTotal + amount;
+    return this;
```

Comment:
why do you return the service itself? It seems to be useful to return the new total.

Reply:
To make multiple add on same action.
Ex: `object.addAmount(10).addAmount(20)...`